src/components/structures/RightPanel.tsx

```diff
@@ -255,14 +261,16 @@ export default class RightPanel extends React.Component<IProps, IState> {
                 break;
 
             case RightPanelPhases.ThreadPanel:
-                card = (
-                    <ThreadPanel
-                        roomId={roomId}
-                        resizeNotifier={this.props.resizeNotifier}
-                        onClose={this.onClose}
-                        permalinkCreator={this.props.permalinkCreator}
-                    />
-                );
+                if (roomId) {
```

Comment:
```suggestion
                if (!!roomId && !!this.props.permalinkCreator) {
```

(to fix implicit boolean cast, and type failure on permalinkCreator)

Reply:
I specifically didn't touch permalinkCreator as there are multiple related type fails around the code and it may be intentional that it can be passed as undefined